### PR TITLE
Quick memory optimisations

### PR DESCRIPTION
**Pull request by mauch**
These changes stop the quadrupling of dataframe in memory when using `iterrows` before upload to the database. There are also a few other minor changes to squeeze out a bit more memory at no cost to processing time.

It runs for me through forced extraction with ~2700 images - and through the measurment pairs with ~1000 images. The pairs dataframe just gets too big after 1000 images.

These changes should be good to hold over a 1.x release (no changes to the databse structure etc.) until further improvements in v2.x release which won't be backwards compatible.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +25,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 #### List of PRs
 
+- [#776](https://github.com/askap-vast/vast-pipeline/pull/776): Minor memory optimisations
```
Comment (ddobie) [inline CHANGELOG.md:28]:
```suggestion
- [#776](https://github.com/askap-vast/vast-pipeline/pull/776): fix: Minor memory optimisations
```

Although I realise that I didn't adhere to this formatting in the previously listed PR!!

File: vast_pipeline/pipeline/forced_extraction.py [hunk 2]
```diff
@@ -413,11 +413,10 @@ def image_data_func(image_name: str) -> Dict[str, Any]:
...
     # compute the rest of the columns
-    intermediate_df = (
-        db.from_sequence(intermediate_df)
-        .map(lambda x: finalise_forced_dfs(**x))
-        .compute()
-    )
+    intermediate_df = list(map(
+        lambda x: finalise_forced_dfs(**x),
+        intermediate_df
+        ))
```
Comment (ddobie) [inline vast_pipeline/pipeline/forced_extraction.py:419]: Maybe add a comment here explaining the logic behind the change, just so someone else doesn't come along and try to switch it back to using dask

File: vast_pipeline/pipeline/forced_extraction.py [hunk 3]
```diff
@@ -479,15 +478,11 @@ def get_fname(n): return os.path.join(
...
     )
-    dfs = list(map(lambda x: (df[df['image'] == x], get_fname(x)), images))
-    n_cpu = cpu_count() - 1
-
-    # writing parquets using Dask bag
-    bags = db.from_sequence(dfs)
-    bags = bags.starmap(
-        lambda df, fname: write_group_to_parquet(df, fname, add_mode))
-    bags.compute(num_workers=n_cpu)
+    dfs = map(lambda x: (df[df['image'] == x], get_fname(x)), images)
```
Comment (ddobie) [inline vast_pipeline/pipeline/forced_extraction.py:481]: Similarly here - add a comment so people don't come back down the line to try and optimise it (and end up making it worse)

File: vast_pipeline/utils/utils.py [hunk 4]
```diff
@@ -392,7 +392,7 @@ def timeStamped(fname, fmt="%Y-%m-%d-%H-%M-%S_{fname}"):
     return datetime.now().strftime(fmt).format(fname=fname)
 
 
-def calculate_n_partitions(df, n_cpu, partition_size_mb=100):
+def calculate_n_partitions(df, n_cpu, partition_size_mb=15):
```
Comment (ddobie) [inline vast_pipeline/utils/utils.py:395]: This also needs a comment/some justification. Dask docs recommend partitions that are ~100MB

**Comment by mauch** [comment]
I've added some comments where you suggested. Also I've modified part of the revalidation step to save ~5GB of RAM for runs with ~2700 images.

**Review by ddobie — APPROVED** [review]
Looks good to me. I haven't run it myself, but you have and it passes tests so I'm happy to approve.

**Pull request merged**